.github/workflows/linux.yml

```diff
@@ -15,520 +10,101 @@ on:
-        run: sh validate.sh -j 2 -w ghc-7.6.3 -v --lib-only -s build
-      - name: Validate lib-tests
-        run: sh validate.sh -j 2 -w ghc-7.6.3 -v --lib-only -s lib-tests
-      - name: Validate lib-suite
-        run: sh validate.sh -j 2 -w ghc-7.6.3 -v --lib-only -s lib-suite
-  validate-8_8_4-old:
-    name: validate.sh old GHCs
-    runs-on: ubuntu-18.04
-    needs: validate-8_8_4
-    container:
-      image: phadej/ghc:8.8.4-xenial
-    steps:
-      - name: System info
-        run: |
-          uname -a
-      - name: Set PATH
-        # https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#adding-a-system-path
-        run: |
-          echo "$HOME/.cabal/bin" >> $GITHUB_PATH
-      - name: Install cabal-plan
-        run: |
-          mkdir -p $HOME/.cabal/bin
-          curl -sL https://github.com/haskell-hvr/cabal-plan/releases/download/v0.6.2.0/cabal-plan-0.6.2.0-x86_64-linux.xz > cabal-plan.xz
-          echo "de73600b1836d3f55e32d80385acc055fd97f60eaa0ab68a755302685f5d81bc  cabal-plan.xz" | sha256sum -c -
-          xz -d < cabal-plan.xz > $HOME/.cabal/bin/cabal-plan
-          rm -f cabal-plan.xz
-          chmod a+x $HOME/.cabal/bin/cabal-plan
-      - name: apt-get update
-        run: apt-get update
-      - name: Install dynamic libraries
-        run: apt-get install -y ghc-8.8.4-dyn
-      - name: Install extra compilers
-        run: apt-get install -y ghc-7.0.4-dyn ghc-7.2.2-dyn ghc-7.4.2-dyn
-      - name: Update Hackage index
-        run: cabal v2-update
-      # https://github.com/actions/checkout/issues/170
-      # - uses: actions/checkout@v2
-      - name: Checkout
-        run: |
-          echo $GITHUB_REF $GITHUB_SHA
-          git clone --depth 1 https://github.com/$GITHUB_REPOSITORY.git .
-          git fetch origin $GITHUB_SHA:temporary-ci-branch
-          git checkout $GITHUB_SHA || (git fetch && git checkout $GITHUB_SHA)
-      - name: Validate print-config
-        run: sh validate.sh -j 2 -w ghc-8.8.4 -v --lib-only -s print-config
-      - name: Validate print-tool-versions
-        run: sh validate.sh -j 2 -w ghc-8.8.4 -v --lib-only -s print-tool-versions
-      - name: Validate build
-        run: sh validate.sh -j 2 -w ghc-8.8.4 -v --lib-only -s build
-      - name: Validate lib-tests
-        run: sh validate.sh -j 2 -w ghc-8.8.4 -v --lib-only -s lib-tests
-      - name: Validate lib-suite
-        run: sh validate.sh -j 2 -w ghc-8.8.4 -v --lib-only -s lib-suite
-      - name: Validate lib-suite-extras --extra-hc /opt/ghc/7.0.4/bin/ghc-7.0.4
-        run: sh validate.sh -j 2 -w ghc-8.8.4 -v --lib-only -s lib-suite-extras --extra-hc /opt/ghc/7.0.4/bin/ghc-7.0.4
-      - name: Validate lib-suite-extras --extra-hc /opt/ghc/7.2.2/bin/ghc-7.2.2
-        run: sh validate.sh -j 2 -w ghc-8.8.4 -v --lib-only -s lib-suite-extras --extra-hc /opt/ghc/7.2.2/bin/ghc-7.2.2
-      - name: Validate lib-suite-extras --extra-hc /opt/ghc/7.4.2/bin/ghc-7.4.2
```

Comment:
I don't see these steps in new CI setup. Is `cabal-install` dropping support for older GHC.

AFAICS the oldest GHC you try to run tests with is GHC-8.0

Is this correct and intentional, @fgaz, @Mikolaj ?

Reply:
Afaics it is not intentional and should be restored.

Reply:
As we are deprecating *build* cabal with ghc < 8.0, all older version should be checked using a newer build (including 7.8 and 7.10) like < 7.6 before this pr

Reply:
the oldest ghc instalable with ghcup is 7.10.3, so we have to continue using apt if we want to keep older ones

Reply:
@phadej It was intentional, following #7534 but now I think I have misunderstood.

We are deprecating building with GHC < 8.0 but we still want to test cabal can use GHC 7.8 and 7.10. Is this correct?

It's a bit of headache (which GHC version we should use to compile the cabal that compiles with what GHC version?) but it can definitely be readded.

Mind adding an issue to track? Please mention me there.

Reply:
@andreabedini: great job! I might have mislead you re #7534. Let's clean up the fallout later on.